lib/pure/terminal.nim

```diff
@@ -448,9 +488,18 @@ type
 
 when not defined(windows):
   var
-    # XXX: These better be thread-local
-    gFG = 0
-    gBG = 0
+    gFG {.threadvar.}: int
+    gBG {.threadvar.}: int
+
+  proc getStyleStr(style: int): string =
```

Comment:
You're losing type safety here. The ``styleCache`` can be changed to an ``array[Style, string]``